### ISSUE TITLE
7.7.0 #B938b54c  Server is unable to authorize account logging in will not complete connection

### ISSUE THREAD

**Issue by pamduncan** [StrangeLoopGames/EcoIssues#9112]
Have had many people complaining that the attempt to log into Stuffed Elk Staging server last night and get stuck at "Connecting.   Asked them to try some other 7.7 servers, some of the ones I have up and ones others are running. For most it occurrred on all servers they tried, a few were able to get on a few, but others were not able to get on those amd got the "Commecting".

In our case after installing 7.7 and players were able to ;log through the server UI (originally loaded using the Network.eco that came with the server install and did not have the new variables in the file) In the Server UI Network Tab  the "Servercategory" was changed from None to Established,   while DetailedDescription and DiscordAddress was give settings in their fields.,  No other changes were made on the server.  Immediately after that players started complaining that they were stuck at the server browser screen with a "Connecting" message but that just stayed and did not go on and connect or  failed.  The Console was showing no login attempts but the Display tab of the Network tab showed login from assorted connection name New Client (number) instead of a player name/account.  After reverting back to the Network.eco which was downloaded with the 7.7 server that does not have the above options in and is the same as the config for for 7.6.3 dome players were then able to connect.  

I am not sure if this happened on the other servers and they configured the 7.7 server changes through the Server UI and then had the same problem or if configuring the server was just coincidental to the problem.  However i was able to change the DiscordAddress and ServerCategory on a 7.7 not having and inssue eartly today and it immediately stopped taking connections.

[![Screenshot from Gyazo](https://gyazo.com/709a2a5cfcbf1c63ec91bb351137c234/raw)](https://gyazo.com/709a2a5cfcbf1c63ec91bb351137c234)

https://gyazo.com/117b5f49dc294407a5ac2208338e3c95 

This happens using Steam install and SLG install,  There are no server or client errors that were found in Dump or Logs folders.   This is a server that has been up and connecting for more that a year, no servers side changes have been made.  Its in a data center and there are neither firewalls or routers to deal with.
Is there anywhere you can point me to that would explain why the server is not seeing a players account information/name when they log on and sees "New Client (number).  Some players are able to log on and connect to server but it very hit or miss and they can get in one time and then be a "New Client (number)" when they try to log back in.

Is there anything you can point me towards to get the server to recognize a players account?

**Comment by johnkslg** [comment]
@BlazeGuy2 can you follow up with Pam to narrow down whats happening?

**Comment by pamduncan** [comment]
This is happening with latest bleeding edge also.  Players logging in the have valid Eco accounts but show up at NewClient (number) whether they had played on server previously or not. Play logging on gets not errors just sits at "Connecting" screen forever

[![Screenshot from Gyazo](https://gyazo.com/3587633a95a57a7c9096154bcfd329eb/raw)](https://gyazo.com/3587633a95a57a7c9096154bcfd329eb)

[![Screenshot from Gyazo](https://gyazo.com/71b7a207314e66efb45d8534f2ce6e0d/raw)](https://gyazo.com/71b7a207314e66efb45d8534f2ce6e0d)
\

Tried: Removing appdata; reinstalling (steam keeps claiming 2 missing files); reinstalling client (steam keeps claiming two missing files). Tried SLG account vs Steam account; tried other accounts.  Is there a configuration change we are missing?

**Comment by BlazeGuy2** [comment]
Can't seem to be able to reproduce this on my machine when I connect to Stuffed Elk or my own server on my machine.  @pamduncan Do you have a remote server that is running 7.7?

**Comment by pamduncan** [comment]
I have several
And its happen on most of them to effected users - sometimes ones that have been in the world before and come back and sometimes to new players that have not been one before.  I was able to get on all these servers before the last 7.7 that dropped to stable and now when I try and log on I am seen as NewClient #### and it will not authenticate m

**Comment by BlazeGuy2** [comment]
@pamduncan Can you send me the save to your big world on 7.7 so that I can try to reproduce it locally for debugging?

**Comment by pamduncan** [comment]
https://www.dropbox.com/s/a8av736uc1ua1jf/Pams%20Big%20World.db?dl=0
https://www.dropbox.com/s/pay7wyru7fem98s/Pams%20Big%20World.eco?dl=0

**Comment by BlazeGuy2** [comment]
@pamduncan After the latest staging build, the issue no longer happens for me.  Try updating the version to what I just uploaded and let me know if this is still happening for you.  Version number will be the same but it is the new build.

**Comment by pamduncan** [comment]
This is what I have found that at least got us connected:  If its happening if you set the server that is not connecting as a favorite and then check the favorite file in appdata and in our cases we found it has a gameport settings that is not the one configured for the server.  When I did this with my server it was linking to the right Ip but it was showing the gameport as 0,  when the gameport was actually set on 26105..  In other cases gameports were set to 3000 and when they Favorited it after not being able to connect it showed the gameport on 37286, in one case is was set on 4000 and the gameport when set as a favorite was 4..  We edited the edited the gameport in the favorites file and put the correct port in and then were able to connect without any issue to the server in question   However if we went back to the browser and used the link on the browser page it would not connect and if we removed it as a favorite and then made it as favorite again the wrong port was used again.   It seems like the browser link had the wrong gameport, but I have no way to check that, but at least making it a favorite and editing the gameport in that favorite file allows us to connect.  Several of these cases were Linux servers and with Linux servers this happens with 7.6.3 too.  On windows it just seems to happen on 7.7
